controller/aks-cluster-config-handler.go

```diff
@@ -627,53 +627,55 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, secretsCache w
 		return config, err
 	}
 
-	downstreamNodePools, err := utils.BuildNodePoolMap(config.Spec.NodePools, config.Spec.ClusterName)
-	if err != nil {
-		return config, err
-	}
+	if config.Spec.NodePools != nil {
```

Comment:
nit: put the `aks.NewAgentPoolClient` call above in this if-block because it isn't needed otherwise.